pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/BaseBooleanAggregationFunction.java

```diff
@@ -233,7 +233,8 @@ public Integer merge(Integer intermediateResult1, Integer intermediateResult2) {
 
   @Override
   public DataSchema.ColumnDataType getIntermediateResultColumnType() {
-    return DataSchema.ColumnDataType.BOOLEAN;
+    // The intermediateResult type is integer. The final result is converted to a boolean.
+    return DataSchema.ColumnDataType.INT;
```

Comment:
this could cause backward-compatibility issue when upgrading. let's make sure this doesn't cause issues (where is this function being used in v1) 

in v2 as long as the expected data type conforms with calcite we are good

Reply:
Got it. 
Addressed this by converting Boolean objects to Integer objects in NewAggregateOperator.